modules/AlphaWalletFoundation/AlphaWalletFoundation/EtherClient/Models/ContractMethodCalls/Erc721TokenUriMethodCall.swift

```diff
@@ -8,11 +8,119 @@
 import Foundation
 import BigInt
 
+public struct Base64Decoder {
+    public typealias Response = (mimeType: String?, encoding: Encoding?, data: Data)
+    public enum Encoding {
+        case base64
+        case other(String)
+
+        init(string: String) {
+            switch string {
+            case "base64":
+                self = .base64
+            default:
+                self = .other(string)
+            }
+        }
+    }
+
+    public init() {}
+    
+    /// for strings like: "data:application/json;base64,eyJuYW1lIjoiQHBvcnRlci5sZW5zIiwiZGVzY3JpcHRpb24iO=="
```

Comment:
Add a test?

Reply:
Also the example doesn't seem to work though it works in https://www.base64decode.org

Reply:
I tried "eyJuYW1lIjoiQHBvcnRlci5sZW5zIiwiZGVzY3JpcHRpb24iO==" and the page returned `{"name":"@porter.lens","description"`

Reply:
example base 64 is just part of what was there, that is ok

Reply:
I don't understand, I mean:

```
let data = Base64Decoder().decode(string: "data:application/json;base64,eyJuYW1lIjoiQHBvcnRlci5sZW5zIiwiZGVzY3JpcHRpb24iO==")
//nil, but should be: {"name":"@porter.lens","description"

let data = Base64Decoder().decode(string: "data:application/json;base64,eyJuYW1lIjoiQHBvcnRlci5sZW5zIiwiZGVzY3JpcHRpb24iOiJzb21ldGhpbmcifQ==")
//nil, but should be: {"name":"@porter.lens","description":"something"}
```

The 2nd is a valid JSON.

Reply:
yes, first example is invalid, i took existed large base64 encoded json and copied smal part of it

Reply:
Hmm.. I might have misunderstood. Would you help add a test case for the function if you haven't? I'll check again.